internal/scheduledupgrades/scheduledupgrades.go

```diff
@@ -0,0 +1,94 @@
+package scheduledupgrades
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/blang/semver"
+	"github.com/giantswarm/microerror"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/apiextensions/v3/pkg/annotation"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+
+	"github.com/giantswarm/azure-admission-controller/internal/errors"
+	"github.com/giantswarm/azure-admission-controller/internal/releaseversion"
+	"github.com/giantswarm/azure-admission-controller/internal/semverhelper"
+)
+
+func ValidateClusterAnnotationUpgradeTime(oldCluster *capi.Cluster, newCluster *capi.Cluster) error {
+	if updateTime, ok := newCluster.GetAnnotations()[annotation.UpdateScheduleTargetTime]; ok {
+		if oldCluster != nil {
+			if updateTimeOld, ok := oldCluster.GetAnnotations()[annotation.UpdateScheduleTargetTime]; ok {
+				if updateTime == updateTimeOld {
+					return nil
+				}
+			}
+		}
+		if !ValidateUpgradeScheduleTime(updateTime) {
+			return microerror.Maskf(notAllowedError,
+				fmt.Sprintf("Cluster annotation '%s' value '%s' is not valid. Value must be in RFC822 format and UTC time zone (e.g. 30 Jan 21 15:04 UTC) and should be a date 16 mins - 6months in the future.",
```

Comment:
where do these rules come from? have we documented it somewhere?

Reply:
This method call comes straight from AWS admission controller. No idea about documentation